gradle/build-scan.gradle

```diff
@@ -76,9 +77,9 @@ buildScan {
         value 'Git Branch', branch
         tag branch
       }
-      if (System.getenv('GIT_COMMIT')) {
-        value 'Git Commit ID', System.getenv('GIT_COMMIT')
-        link 'Source', "https://github.com/elastic/elasticsearch/tree/${System.getenv('GIT_COMMIT')}"
+      value 'Git Commit ID', BuildParams.gitRevision
+      link 'Source', "https://github.com/elastic/elasticsearch/tree/${BuildParams.gitRevision}"
+      if (System.getenv('GIT_PREVIOUS_COMMIT')) {
         background {
           def changes = "git diff --name-only ${System.getenv('GIT_PREVIOUS_COMMIT')}..${System.getenv('GIT_COMMIT')}".execute().text.trim()
```

Comment:
Should the use of GIT_COMMIT here also be replaced with BuildParams.gitRevision?

Reply:
I didn't change this because `GIT_PREVIOUS_COMMIT` is subject to the same "problem". I think if anything we might just want to remove this entirely if it's not guaranteed to be accurate.

Reply:
So yeah, since we can't rely on `GIT_COMMIT` or `GIT_PREVIOUS_COMMIT` from Jenkins being actually correct I've just ditched this entirely.